chatcli_gpt/plugins.py

```diff
@@ -31,33 +39,60 @@ def extract_blocks(response_text, plugin):
     return matches
 
 
+def exec_bash(code):
+    try:
+        result = subprocess.run(
+            ["/bin/bash", "-c", code], capture_output=True, text=True
+        )
+    except Exception:  # pylint: disable=broad-except
+        print(traceback.format_exc())
+
+    return {
+        "result": result.stdout.strip(),
+        "error": result.stderr.strip(),
+    }
+
+
 def exec_python(code):
-    buffer = io.StringIO()
-
-    with contextlib.redirect_stdout(buffer):
-        global_scope = globals()
-        try:
-            mod = ast.parse(code, mode="exec")
-            if isinstance(mod.body[-1], ast.Expr):
-                last_expr = mod.body.pop()
-                exec(compile(mod, "<ast>", "exec"), global_scope)
-                result = eval(compile(ast.Expression(last_expr.value), "<ast>", "eval"), global_scope)
-                if result is not None:
-                    print(result)
-            else:
-                exec(code, global_scope)
-        except Exception:  # pylint: disable=broad-except
-            print(traceback.format_exc())
-
-    return buffer.getvalue().strip()
+    stdout = io.StringIO()
+    stderr = io.StringIO()
+
+    with contextlib.redirect_stdout(stdout):
```

Comment:
I think we can do

```
with contextlib.redirect_stdout(stdout), contextlib.redirect_stderr(stderr):
```
and avoid the extra level of nesting here.

Reply:
learned something new! `with` accepts a tuple of contexts, that's sweet.